cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -165,6 +168,18 @@ protected void configure() {
 
   @Override
   public void initialize(WorkflowContext context) throws Exception {
+    MDC.put(Constants.PIPELINE_LIFECYCLE_TAG, Constants.PIPELINE_LIFECYCLE_TAG_VALUE);
```

Comment:
You might want to refactor the `LocationAwareSamplingLogger` to do that in order to maintain the original log location.